### ISSUE TITLE
dark theme for Codidact

### ISSUE THREAD

**Issue by Istiakshovon** [codidact/qpixel#565]
I am tired of using extension. When I use extension (dark theme) for Codidact and, move to a dark theme website then that dark website becomes lighter. Which I totally hate. I always have to change theme for changing tabs. So, I am thinking to build dark theme for Codidact. In SE sites, only SO has dark theme (maybe that's in beta mode). 

Reason of building dark theme : 

- Not to get distracted while writing big answers/questions
- Not to affect eyes.

**Comment by cellio** [comment]
We've gotten other requests for a dark theme, so if somebody wants to work on one that sounds like a benefit.  I don't know a lot about front-end stuff; is this something that should be done at the CoDesign layer?

**Comment by Istiakshovon** [comment]
> We've gotten other requests for a dark theme, so if somebody wants to work on one that sounds like a benefit. I don't know a lot about front-end stuff; is this something that should be done at the CoDesign layer?

I am sure of CoDesign. Actually, I have to design scss file two way (I could let them in a single page). If `background.light{}` and `background.dark{}`. If someone chooses dark theme than, `.light` won't show to him. That's the simple way to do that. But, there's huge scss file that's why it will take longer to be done. I will work on it with patience. :)

**Comment by cellio** [comment]
Please talk with @luap42 about the best way to make this work with CoDesign.  There are some color things already in there; a dark mode would need to play well with them (or they'd need to be adjusted), and there might already be hooks in there for the things you want to do.

**Comment by ArtOfCode-** [comment]
Yes, this is something that should be done from Co-Design. There'll be some things that need to be done in QPixel CSS as well, but most of it should come from CD. Remember there's a media query for this: `@media screen and (prefers-color-scheme: dark) { }`

**Comment by cellio** [comment]
CoDesign repo: https://github.com/codidact/co-design

**Comment by razetime** [comment]
@ArtOfCode- would be worth transferring over to co-design, then?

**Comment by Istiakshovon** [comment]
@razetime some design(header widget and some more) had been in co-design. Those designs don't let me do any changes to qpixel

**Comment by Istiakshovon** [comment]
![Screenshot from 2021-08-02 03-14-21](https://user-images.githubusercontent.com/58106197/127819393-3d824aaf-b4ec-401c-a8ff-f37992c2d8bb.png)

**Comment by cellio** [comment]
Sounds like this needs changes in both CoDesign and qpixel.  Could you create an issue in CoDesign to track it and link the two of them together?  Thanks.

**Comment by Istiakshovon** [comment]
https://github.com/codidact/co-design/issues/57

**Comment by cellio** [comment]
When this happens, we should status-tag this: https://meta.codidact.com/posts/275840#answer-275840.

**Comment by Istiakshovon** [comment]
We use dark theme for rest our eyes. Recently, I was reading some PDF files. Then, I realized my eyes are getting hurt when  I read a book for long moments. Even, I was using Dark Reader for too long for websites. Even, some time I stay in a single website for too long time. Later, I noticed I should have a better way to rest my eyes. I had decreased contrast then, I noticed it working very well now. 

So, I am thinking if we add contrast (User will increase or decrease contrast as their needs) than, it might get better. Contrast is very helpful for eyes either. I want someone else's opinion. 

<details>
<summary>With 84% contrast</summary>

![Screenshot from 2021-08-24 00-07-35](https://user-images.githubusercontent.com/58106197/130471294-0ef88e04-b8e2-4b0a-9182-106d7ebf3b8e.png)

</details>

<details>
<summary>Without contrast</summary>

![Screenshot from 2021-08-24 00-08-07](https://user-images.githubusercontent.com/58106197/130471406-e61104d6-3cdf-4f38-bf06-1e75ae6c3850.png)

</details>
